src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs

```diff
@@ -254,7 +274,7 @@ public async Task<KeyOperationResult> SetMasterKeyAsync(string value = null)
                 if (value == null)
                 {
                     // Generate a new secret (clear)
-                    masterKey = GenerateSecret();
+                    masterKey = GenerateIdentifiableSecret(MasterKeySeed);
```

Comment:
>GenerateIdentifiableSecret [](http://example.com/codeflow?start=32&length=26)

Assuming we retain a distinct seed for master keys, we should consider a distinct helper GenerateMasterKey or similar, which would ensure that the master key seed is always utilized. I did not make this change, though, wanting to get feedback from you on whether the grouping I've proposed here, master, system, and function keys, makes sense.